### PR TITLE
3 ➡️ 4

### PR DESCRIPTION
**Pull request by chapulina**
# ➡️ Forward port

Port `ign-gazebo3` to `ign-gazebo4`

Branch comparison: https://github.com/ignitionrobotics/ign-gazebo/compare/ign-gazebo4...ign-gazebo3

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

**Comment by chapulina** [comment]
@adlarkin , mind checking if I handled #1149 correctly? Thanks!

File: src/EntityComponentManager.cc [hunk 1]
```diff
@@ -1313,53 +1315,34 @@ void EntityComponentManager::SetState(
...
-        // component so that we don't get here, gotta figure out why this
-        // doesn't update the component. The following line prints the correct
-        // values.
-        // igndbg << *comp << "  " << *newComp.get() << std::endl;
-        // *comp = *newComp.get();
-
-        // When above TODO is addressed, uncomment AddModifiedComponent below
-        // unless calling SetChanged (which already calls AddModifiedComponent)
-        // this->dataPtr->AddModifiedComponent(entity);
+        this->dataPtr->AddModifiedComponent(entity);
```
Comment (adlarkin) [inline src/EntityComponentManager.cc:1345]: This marks the component as changed, but the component's data needs to be updated before marking it as changed. Adding the following lines before the `AddModifiedComponent` call should update the component's data:
```c++
        std::istringstream istr(compMsg.component());
        comp->Deserialize(istr);
```

Comment (chapulina) [reply src/EntityComponentManager.cc:1347]: Thanks! Fixed in 546c95f . I just realized I could have looked at #1131 :grimacing:

**Review by adlarkin — APPROVED** [review]
Thanks for taking care of the forward port!

**Pull request merged**